internal/target/repository.go

```diff
@@ -237,52 +237,69 @@ func (r *Repository) FetchAuthzProtectedEntitiesByScope(ctx context.Context, pro
 func (r *Repository) listTargets(ctx context.Context, opt ...Option) ([]Target, error) {
 	const op = "target.(Repository).listTargets"
 
-	if len(r.permissions) == 0 {
-		return []Target{}, nil
+	opts := GetOpts(opt...)
+	limit := r.defaultLimit
+	if opts.WithLimit != 0 {
+		limit = opts.WithLimit
 	}
 
-	where, args := r.listPermissionWhereClauses()
```

Comment:
This PR removes the use of the list permission where clause in favor of simplifying the querying and performing this filtering later. This was added as a stop gap to avoid requesting results that were never going to be shown to the user at a time when we were reading a whole table at once, and shouldn't be necessary anymore. It would still allow us to avoid requesting items that the user doesn't have access to, but I'd be worried about the impact on the querying performance given that we're very carefully trying to make Postgres use the indexes we've created. Feedback appreciated? Should we keep this? How can we do so without affecting the querying performance?

Reply:
We should not remove this. It wasn't really added as a stop gap. It was a notable performance improvement by moving the filtering *into* the query where it would be more performant. Moving the filtering out of the query is far more likely to be less performant. We should examine some query plans to see how the proposed indexes are used. We can leverage CTEs to help guide postgres if need be, but more often there would be good reason for postgres to avoid using an index.

Reply:
OK cool, we'll add it back. I'll try to weave it into the CTE we're using to pick the rows rather than the wrapping query that uses the view, but feedback would be appreciate on how to best construct the query.